compaction/compaction.cc

```diff
@@ -277,6 +277,18 @@ class compacted_fragments_writer {
     stop_func_t _stop_compaction_writer;
     std::optional<utils::observer<>> _stop_request_observer;
     bool _unclosed_partition = false;
+    struct partition_state {
+        dht::decorated_key_opt dk;
+        // Partition tombstone is saved for the purpose of replicating it to every fragment storing a partition pL.
+        // Then when reading from the SSTable run, we won't unnecessarily have to open >= 2 fragments, the one which
+        // contains the tombstone and another one(s) that has the partition slice being queried.
+        ::tombstone tombstone;
+        // Used to determine whether any active tombstones need closing at EOS.
+        ::tombstone current_emitted_tombstone;
+        // Track last emitted clustering row, which will be used to close active tombstone if splitting partition
+        position_in_partition last_pos = position_in_partition(position_in_partition::end_of_partition_tag_t());
```

Comment:
Better initialize to before all culstered rows.

Reply:
> Almost there. Missing re-admission of active tombstone. I wonder how the test didn't catch this.

Probably the closing done by mutation_compactor was ignored by sstable_writer, as from the sstable_writer perspective there was no active tombstone. I need to extend the unit test to validate the range tombstones stored across the fragments.